tests/unit/utils/test_urlutils.py

```diff
@@ -531,22 +531,36 @@ class TestIncDecNumber:
 
     """Tests for urlutils.incdec_number()."""
 
-    @pytest.mark.parametrize('url, incdec, output', [
-        ("http://example.com/index1.html", "increment", "http://example.com/index2.html"),
-        ("http://foo.bar/folder_1/image_2", "increment", "http://foo.bar/folder_1/image_3"),
-        ("http://bbc.c0.uk:80/story_1", "increment", "http://bbc.c0.uk:80/story_2"),
-        ("http://mydomain.tld/1_%C3%A4", "increment", "http://mydomain.tld/2_%C3%A4"),
-        ("http://example.com/site/5#5", "increment", "http://example.com/site/6#5"),
-
-        ("http://example.com/index10.html", "decrement", "http://example.com/index9.html"),
-        ("http://foo.bar/folder_1/image_3", "decrement", "http://foo.bar/folder_1/image_2"),
-        ("http://bbc.c0.uk:80/story_1", "decrement", "http://bbc.c0.uk:80/story_0"),
-        ("http://mydomain.tld/2_%C3%A4", "decrement", "http://mydomain.tld/1_%C3%A4"),
-        ("http://example.com/site/5#5", "decrement", "http://example.com/site/4#5"),
+    @pytest.mark.parametrize('url, incdec, output, segments', [
+        ("http://example.com/index1.html", "increment",
+         "http://example.com/index2.html", {'path'}),
+        ("http://foo.bar/folder_1/image_2", "increment",
+         "http://foo.bar/folder_1/image_3", {'path'}),
+        ("http://bbc.c0.uk:80/story_1", "increment",
+         "http://bbc.c0.uk:80/story_2", {'path'}),
+        ("http://mydomain.tld/1_%C3%A4", "increment",
+         "http://mydomain.tld/2_%C3%A4", {'path'}),
+        ("http://example.com/site/5#5", "increment",
+         "http://example.com/site/6#5", {'path'}),
+        ("http://example.com/site/1#1", "increment",
+         "http://example.com/site/1#2", {'path', 'anchor'}),
+        ("http://example.com/site/1?page=2#3", "increment",
+         "http://example.com/site/1?page=3#3", {'path', 'query'}),
+
+        ("http://example.com/index10.html", "decrement",
+         "http://example.com/index9.html", {'path'}),
+        ("http://foo.bar/folder_1/image_3", "decrement",
+         "http://foo.bar/folder_1/image_2", {'path'}),
+        ("http://bbc.c0.uk:80/story_1", "decrement",
+         "http://bbc.c0.uk:80/story_0", {'path'}),
+        ("http://mydomain.tld/2_%C3%A4", "decrement",
+         "http://mydomain.tld/1_%C3%A4", {'path'}),
+        ("http://example.com/site/5#5", "decrement",
+         "http://example.com/site/4#5", {'path'}),
```

Comment:
I think some tests for the regex above are missing, e.g. with a segment like those:

* `123foo` (number at start)
* `12bar23` (multiple numbers)
* `foo12bar` (text before/after number preserved?)